pyproject.toml

```diff
@@ -1,25 +1,57 @@
+[build-system]
+requires = ["setuptools >= 68.0", "setuptools_scm>=8"]
+build-backend = "setuptools.build_meta"
+
 [project]
 name = "mreg"
-version = "0.0.1"
+requires-python = ">=3.8"
+dependencies = [
+    "Django>4",
+    "djangorestframework==3.14.0",
+    "django-auth-ldap==4.8.0",
+    "django-logging-json==1.15",
+    "django-netfields==1.3.2",
+    "django-filter==24.2",
+    "structlog==24.1.0",
+    "rich==13.7.1",
+    "gunicorn==22.0.0",
+    "idna==3.7",
+    "psycopg2-binary==2.9.9",
+    "pika==1.3.2",
+    "sentry-sdk==2.3.1",
+    "tzdata==2024.1",
```

Comment:
Ideally we don't lock dependency versions here. That is what the lock file is for. These should be more permissive, so that we don't run into a dependency resolution problem.

I.e. multiple dependencies rely on `idna`, but in upgrading one of them, it bumps its own `idna` requirement to  `idna>=3.8`. That would create a dependency resolution conflict, since we require `idna==3.7` specifically.

The solution is to change all these from `==` to `>=`.

Reply:
I agree. Do we trust people to understand semantic versioning enough to just do major versions, or do we do major.minor? I'd ideally like to see major only, but I feel very confident in that something will break (see DRF).

Reply:
I think it's fine either way as long as we always use the lock file to install dependencies, and we are deliberate with our dependency upgrades.

We can upgrade each dependency individually with `uv sync --upgrade-package <package>`.

If we _know_, for example, Django 6 is broken, we can upgrade with `uv sync --upgrade-package "Django>=5, <6"` to ensure we don't upgrade to an incompatible version. 

There are probably some nuances here I'm missing, but the point is that _we_ control when we want to actually upgrade a package. The more relaxed version specifiers just avoids dependency resolution conflicts when we want to do so.

Reply:
Sounds good. `>=` it is!